### PR TITLE
Extend Spring-rabbitmq coverage

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://github.com/apache/camel-quarkus/issues/6362

### CODE REVIEW

File: integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqUtil.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Very simple serialization help to avoid dependencies for json serialization.
+ */
+public class SpringRabbitmqUtil {
```
Comment (jamesnetherton) [inline integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqUtil.java:27]: Should be `final` and have `private` constructor.

File: integration-tests/spring-rabbitmq/src/test/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqTest.java [hunk 2]
```diff
@@ -39,69 +54,350 @@ class SpringRabbitmqTest {
     private final static String ROUTING_KEY_POLLING = "pollingKey";
     private ConnectionFactory connectionFactory;
 
+    /**
+     * <a:href link=
+     * "https://camel.apache.org/components/4.4.x/spring-rabbitmq-component.html#_default_exchange_name">DEFAULT EXCHANGE
+     * NAME</a:href>
+     */
```
Comment (jamesnetherton) [inline integration-tests/spring-rabbitmq/src/test/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqTest.java:61]: Not sure if `<a:href>` is valid HTML / JavaDoc?

Maybe just use the plain URL or remove the comment entirely. Likewise for the other JavaDoc blocks on the other test headings.

File: integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqResource.java [hunk 3]
```diff
@@ -40,51 +47,114 @@ public class SpringRabbitmqResource {
...
+                }
+
+                return Response.ok(SpringRabbitmqUtil.listToString(results)).build();
+            }
+
+            return Response.ok(consumerTemplate.receiveBody(directName, _timeout, String.class)).build();
+        } catch (RuntimeCamelException e) {
+            StringWriter sw = new StringWriter();
+            e.printStackTrace(new PrintWriter(sw));
+            return Response.status(500).entity(sw.toString()).build();
```
Comment (jamesnetherton) [inline integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqResource.java:109]: Do you actually need to send the stack traces back to the client?

IIRC Camel's ErrorHandler should log the exceptions.

Comment (JiriOndrusek) [reply integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqResource.java:109]: I haven find a better way f doing so, the error (which I'd like to assert) is about 3 exceptions deep. Perhaps I can search for the proper Exception in the Resource and may return only relevant part of the text, so the ammount af data is not large. WDYT?

Comment (JiriOndrusek) [reply integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqResource.java:109]: Let me recheck the code and hopefully I can find a better solution. I agree that thias is not the ideal way of achieving so simple assert.

File: integration-tests/spring-rabbitmq/src/test/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqTest.java (hunk 2 above)
Comment (JiriOndrusek) [reply integration-tests/spring-rabbitmq/src/test/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqTest.java:61]: It should be `<a href` 

But I'll remove the links, as the  test methods could be easily linked to the chapters via its names

File: integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqRouteBuilder.java [hunk 4]
```diff
@@ -26,17 +29,75 @@ public class SpringRabbitmqRouteBuilder extends RouteBuilder {
...
+        from("spring-rabbitmq:exchange-for-topic?queues=queue-for-topicA&routingKey=topic.#&connectionFactory=#connectionFactory&autoDeclare=true&exchangeType=topic")
+                .transform(body().prepend("Hello from topic: "))
+                .to("direct:topic");
+
+        //manual acknowledgement
+        from("spring-rabbitmq:exchange-for-manual-ack?queues=queue-for-manual-ack&routingKey=key-for-manual-ack&connectionFactory=#connectionFactory&autoDeclare=true&acknowledgeMode=MANUAL")
+                .process(exchange -> {
+                    //simulate processing time 20 s (has to be bigger than timeout to read from direct routes
+                    // -> see SpringRabbitmqResource.getFromDirect (5 seconds)
+                    TimeUnit.SECONDS.sleep(20);
```
Comment (aldettinger) [inline integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqRouteBuilder.java:64]: Those comments are super useful :+1:

File: integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqResource.java (hunk 3 above)
Comment (jamesnetherton) [reply integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqResource.java:109]: Ah, sorry. I should have read the assertions more closely. Feel free to leave it as-is then.

IIRC you can set the response entity to the exception object and RestEasy will attempt to convert it (maybe to JSON). But I wouldn't spend time adjusting the code just for that.

File: integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqRouteBuilder.java [hunk 5]
```diff
@@ -26,17 +29,75 @@ public class SpringRabbitmqRouteBuilder extends RouteBuilder {
...
+
+        //deadletter
+        from("spring-rabbitmq:exchange-for-deadletter?queues=queue-for-deadletter&routingKey=routing-key-for-deadletter&connectionFactory=#connectionFactory&autoDeclare=true&deadLetterExchange=exchange-for-deadletter-DL&deadLetterExchangeType=fanout&deadLetterQueue=exchange-for-deadletter-DL&deadLetterRoutingKey=any-key&rejectAndDontRequeue=true")
+                .process(exchange -> {
+                    //forced exception
+                    throw new RuntimeException("forced exception to trigger dead letter exchange");
+                })
+                .to("direct:deadletter");
+
+        //redirection from deadletter queu to direct (with autoDeclare = false,because the bindig is created by above route)
```
Comment (aldettinger) [inline integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqRouteBuilder.java:80]:
```suggestion
        //redirection from deadletter queu to direct (with autoDeclare = false,because the binding is created by above route)
```

File: integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqUtil.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+    public static String headersToString(Map<String, Object> headers) {
```
Comment (aldettinger) [inline integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqUtil.java:29]: Maybe some of those methods are already implemented:
 * apache commStringUtils.join() ?
 * java.util.String.join() ?
 * StringUtils.splitPreserveAllTokens() ?
 *

**Review by aldettinger — APPROVED** [review]

File: integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqUtil.java (hunk 1 above)
Comment (JiriOndrusek) [reply integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqUtil.java]: fixed

File: integration-tests/spring-rabbitmq/src/test/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqTest.java (hunk 2 above)
Comment (JiriOndrusek) [reply integration-tests/spring-rabbitmq/src/test/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqTest.java]: I removed the comments

File: integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqResource.java (hunk 3 above)
Comment (JiriOndrusek) [reply integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqResource.java]: I found that I don't assert those stack traces anymore, so I removed that code completely.

File: integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqRouteBuilder.java (hunk 5 above)
Comment (JiriOndrusek) [reply integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqRouteBuilder.java]: fixed

File: integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqRouteBuilder.java (hunk 4 above)
Comment (JiriOndrusek) [reply integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqRouteBuilder.java:64]: Thanks!

File: integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqUtil.java (hunk 6 above)
Comment (JiriOndrusek) [reply integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqUtil.java]: (I use those transformations only for constant input, therefore no special corner case should happen)

**Review by jamesnetherton — APPROVED** [review]
